src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,83 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
 import './FindMovie.scss';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieCard } from '../MovieCard';
+
+const PREVIEW_IMG = 'https://via.placeholder.com/360x270.png?text=no%20preview';
+
+type Props = {
+  movies: Movie[];
+  setMovies: (prevMovies: (prevMovies: Movie[]) => Movie[]) => void;
+};
+
+export const FindMovie: React.FC<Props> = ({
+  movies,
+  setMovies,
+}) => {
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [title, setTitle] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  function resetForm() {
+    setMovie(null);
+    setTitle('');
+    setIsError(false);
+  }
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setIsError(false);
+  };
+
+  const handleAddClick = () => {
+    if (movie && !movies.some(m => m.imdbId === movie.imdbId)) {
+      setMovies(prevMovies => ([...prevMovies, movie]));
+    }
+
+    resetForm();
+  };
+
+  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setLoading(true);
+
+    getMovie(title)
+      .then((response) => {
+        if ('Title' in response) {
+          const {
+            Title, Plot, Poster, imdbID,
+          } = response;
+
+          const newMovie: Movie = {
+            title: Title,
+            description: Plot,
+            imgUrl: Poster !== 'N/A' ? Poster
+              : PREVIEW_IMG,
+            imdbUrl: `https://www.imdb.com/title/${imdbID}`,
+            imdbId: imdbID,
+          };
+
+          setMovie(newMovie);
+        } else {
+          setIsError(true);
+        }
+      })
+      .catch(() => setIsError(true))
```

Comment:
Whenever you'd like to throw an error and then catch this error. You can do like this

```
.then(res = > {
   if (res.ok) {
     return res;
   }
   throw Error('my_error_message')
})
.catch(e => setError(e.message))
```

Reply:
now I understood, thank you!